numpy/lib/tests/test_financial.py

```diff
@@ -147,6 +215,79 @@ def test_when(self):
+                                   Decimal('0'), Decimal('1')),
+                            np.pmt(Decimal('0.08')/Decimal('12'), Decimal('5')*Decimal('12'), Decimal('15000.'),
+                                   Decimal('0'), 'begin'), 2)
+        # end
+        assert_almost_equal(np.pmt(Decimal('0.08')/Decimal('12'), Decimal('5')*Decimal('12'), Decimal('15000.'),
+                                   Decimal('0')),
+                            np.pmt(Decimal('0.08')/Decimal('12'), Decimal('5')*Decimal('12'), Decimal('15000.'),
+                                   Decimal('0'), 'end'), 2)
+        assert_almost_equal(np.pmt(Decimal('0.08')/Decimal('12'), Decimal('5')*Decimal('12'), Decimal('15000.'),
+                                   Decimal('0'), Decimal('0')),
+                            np.pmt(Decimal('0.08')/Decimal('12'), Decimal('5')*Decimal('12'), Decimal('15000.'),
+                                   Decimal('0'), 'end'), 2)
+
+        # begin
+        assert_almost_equal(np.ppmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('60'), Decimal('55000'),
+                                    Decimal('0'), Decimal('1')),
+                            np.ppmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('60'), Decimal('55000'),
+                                    Decimal('0'), 'begin'), 2)
+        # end
+        assert_almost_equal(np.ppmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('60'), Decimal('55000'),
+                                    Decimal('0')),
+                            np.ppmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('60'), Decimal('55000'),
+                                    Decimal('0'), 'end'), 2)
+        assert_almost_equal(np.ppmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('60'), Decimal('55000'),
+                                    Decimal('0'), Decimal('0')),
+                            np.ppmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('60'), Decimal('55000'),
+                                    Decimal('0'), 'end'), 2)
+
+        # begin
+        assert_almost_equal(np.ipmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('24'), Decimal('2000'),
+                                    Decimal('0'), Decimal('1')).flat[0],
+                            np.ipmt(Decimal('0.1') / Decimal('12'), Decimal('1'), Decimal('24'), Decimal('2000'),
+                                    Decimal('0'), 'begin').flat[0], 2)
+        # end
+        assert_almost_equal(np.ipmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('24'), Decimal('2000'),
+                                    Decimal('0')).flat[0],
+                            np.ipmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('24'), Decimal('2000'),
+                                    Decimal('0'), 'end').flat[0], 2)
+        assert_almost_equal(np.ipmt(Decimal('0.1')/Decimal('12'), Decimal('1'), Decimal('24'), Decimal('2000'),
```

Comment:
Does  `assert_almost_equal` cast the decimal to float first? And what's the point of using `Decimal` if you're only going to check for approximate equality?

Reply:
I suppose I could change them much like I did in one of the test to check the specific values. 

As for the approximate equality, that is because of the `assert_almost_equal`

Is it alright if I make the class inherit from `unittest.TestCase` and use the `self.assertEquals`? that's really what I want anyway. I was avoiding adding too much new. 

I could also use `assert` directly, which will raise the assertion error, but won't have nice messaging when the test fails by default.

Reply:
We will be getting rid of `unittest` in the switch to pytest.